apps/rush-lib/src/api/VersionMismatchFinder.ts

```diff
@@ -7,14 +7,139 @@ import { RushConfiguration } from './RushConfiguration';
 import { RushConstants } from '../logic/RushConstants';
 import { PackageJsonDependency, DependencyType, PackageJsonEditor } from './PackageJsonEditor';
 import { CommonVersionsConfiguration } from './CommonVersionsConfiguration';
+import { RushConfigurationProject } from './RushConfigurationProject';
 
-export interface IVersionMismatchFinderProject {
-  packageName: string;
+export interface IVersionMismatchFinderEntityOptions {
+  friendlyName: string;
   cyclicDependencyProjects: Set<string>;
-  packageJsonEditor: PackageJsonEditor;
   skipRushCheck?: boolean;
 }
 
+export abstract class VersionMismatchFinderEntity {
```

Comment:
Even though this source file is not too big, it's accumulating a bunch of types.  Also, it's not really "API" as far as I can tell.

It might be a good idea to move the file under **src/logic/versionMismatch** and split it into some smaller files.

(That could be a separate PR.)

Reply:
I'm good with that. I'll move things around tomorrow.